orangecontrib/text/corpus.py

```diff
@@ -100,6 +100,22 @@ def extend_corpus(self, metadata, Y):
         self.X = self.W = np.zeros((len(self), 0))
         Table._init_ids(self)
 
+    def update_attributes(self, X, dictionary):
+        """
+        Adds X to the corpus and updates its domain accordingly.
+        """
+        self.X = np.hstack((self.X, X))
+
+        dict_key_count = len(sorted(dictionary.keys()))
+        features = [dictionary[key] for key in range(dict_key_count)]
+        attr = [ContinuousVariable.make(f) for f in features]
+        new_domain = Domain(
+                attributes=attr,
```

Comment:
@nikicc I've updated the method, so that we do not lose the attributes. There is a thing we should consider though. Now it's possible to connect the BoW result into another BoW, that duplicates the attributes and their respective values :D Not exactly a likely use case, but these are hard to predict. If this is fine at this point, then I will proceed with the push, otherwise, we can find some other solution.